src/Symfony/Component/Env/Dotenv.php

```diff
@@ -0,0 +1,365 @@
+            if ($this->export) {
+                throw new FormatException('Unable to unset an environment variable', $this->createFormatExceptionContext());
+            }
+
+            throw new FormatException('Missing = in the environment variable declaration', $this->createFormatExceptionContext());
+        }
+
+        if (' ' === $this->data[$this->cursor]) {
+            throw new FormatException('Whitespace are not supported after the variable name', $this->createFormatExceptionContext());
+        }
+
+        if ('=' !== $this->data[$this->cursor]) {
+            throw new FormatException('Missing = in the environment variable declaration', $this->createFormatExceptionContext());
+        }
+        ++$this->cursor;
+
+        return $matches[1];
+    }
+
+    private function lexValue()
+    {
+        if ("\n" === $this->data[$this->cursor]) {
+            $this->skipEmptyLines();
+
+            return '';
+        }
+
+        if (preg_match('/ *(?!#.*)?(?:\n|$)/Am', $this->data, $matches, null, $this->cursor)) {
+            $this->moveCursor($matches[0]);
+
+            return '';
+        }
+
+        if (' ' === $this->data[$this->cursor]) {
+            $this->skipComment();
+
+            // not a problem if the value is only a comment and/or whitespace
+            if ($this->cursor === $this->end || "\n" === $this->data[$this->cursor - 1]) {
+                return '';
+            }
+
+            throw new FormatException('Whitespace are not supported before the value', $this->createFormatExceptionContext());
+        }
+
+        $value = '';
+        $singleQuoted = false;
+        $notQuoted = false;
+        if ("'" === $this->data[$this->cursor]) {
+            $singleQuoted = true;
+            ++$this->cursor;
+            while ("\n" !== $this->data[$this->cursor]) {
+                if ("'" === $this->data[$this->cursor]) {
+                    if ($this->cursor + 1 === $this->end) {
+                        break;
+                    }
+                    if ("'" !== $this->data[$this->cursor + 1]) {
+                        break;
+                    }
+
+                    ++$this->cursor;
+                }
+                $value .= $this->data[$this->cursor];
+                ++$this->cursor;
+
+                if ($this->cursor === $this->end) {
+                    throw new FormatException('Missing quote to end the value', $this->createFormatExceptionContext());
+                }
+            }
+            if ("\n" === $this->data[$this->cursor]) {
+                throw new FormatException('Missing quote to end the value', $this->createFormatExceptionContext());
+            }
+            ++$this->cursor;
+            $this->skipComment();
+        } elseif ('"' === $this->data[$this->cursor]) {
+            ++$this->cursor;
+            while ('"' !== $this->data[$this->cursor] || ('\\' === $this->data[$this->cursor - 1] && '\\' !== $this->data[$this->cursor - 2])) {
```

Comment:
Shouldn't we handle escaping? Or it doesn't worst it?
- "Foo" => OK
- "Foo\" => KO
- "Foo\\" => OK
- "Foo\\\" => KO <= this (edge) case look not handled by your implementation.